src/contraction_utils.cpp

```
@@ -463,6 +459,8 @@ bool ordering_data_to_contraction_ordering(
   if (!error_msg.empty()) {
     std::cerr << "Parsing failed on line: \"" << line
               << "\" with error: " << error_msg << std::endl;
+    // throw ERROR_MSG("Parsing failed on line: '", line, "' with error: ",
```

Why do we return `pair<bool, string>` from this function? I feel like returning `void` and `throw`ing any errors is cleaner - the caller can `try-catch` to discard the error if it shouldn't cause a failure.

@s-mandra, your thoughts?

---

Tangentially related: we should either delete the log line or delete the comment here, depending on the decision for this.

---

> Why do we return `pair<bool, string>` from this function? I feel like returning `void` and `throw`ing any errors is cleaner - the caller can `try-catch` to discard the error if it shouldn't cause a failure.
> 
> @s-mandra, your thoughts?

Mhh .. let me say that it would be strange to have a function called `IsOrderingValid` which returns `void` and `throw` in case it isn't. I also agree that it would be cleaner to avoid a pair `{bool, string}` to catch this error. I'm ok with either ways, but we need to adjust few tests because they rely on the `{bool, string}` pair.

---

I'm actually following @95-martin-orion suggestion.

---

> ...it would be strange to have a function called `IsOrderingValid` which returns `void` and `throw` in case it isn't.

Maybe rename to `ValidateOrdering` or similar?

---

> > ...it would be strange to have a function called `IsOrderingValid` which returns `void` and `throw` in case it isn't.
> 
> Maybe rename to `ValidateOrdering` or similar?

Yeah .. `ValidateOrdering` sounds better than `IsOrderingValid`. Let me change it.